src/test/java/com/puppycrawl/tools/checkstyle/internal/utils/TestUtil.java

```diff
@@ -268,4 +263,132 @@ public static int adjustFlushCountForOutputStreamClose(int flushCount) {
+            throw new IllegalStateException(message, ex);
+        }
+    }
+
+    /**
+     * Writes the value of a field using reflection. This method will traverse the
+     * super class hierarchy until a field with name {@code fieldName} is found.
+     *
+     * @param instance the instance whose field to modify
+     * @param fieldName the name of the field
+     * @param value the new value of the field
+     * @throws RuntimeException if the field  can't be changed
+     */
+    public static void setInternalState(Object instance, String fieldName, Object value) {
+        try {
+            final Field field = getClassDeclaredField(instance.getClass(), fieldName);
+            field.set(instance, value);
+        }
+        catch (ReflectiveOperationException ex) {
+            final String message = String.format(Locale.ROOT,
+                    "Failed to set field '%s' for instance of class '%s'",
+                    fieldName, instance.getClass().getSimpleName());
+            throw new IllegalStateException(message, ex);
+        }
+    }
+
+    /**
+     * Invokes a private method for an instance.
+     *
+     * @param instance the instance whose method to invoke
+     * @param methodToExecute the name of the method to invoke
+     * @param arguments the optional arguments
+     * @param <T> the type of the result
+     * @return the method's result
+     * @throws Exception if the method failed
+     * @noinspection ProhibitedExceptionThrown, ThrowInsideCatchBlockWhichIgnoresCaughtException,
+     *               unchecked
+     */
+    public static <T> T invokeMethod(Object instance,
+            String methodToExecute, Object... arguments) throws Exception {
+        final Class<?> clss = instance.getClass();
+        final Method method = getClassDeclaredMethod(clss, methodToExecute, arguments.length);
+        try {
+            return (T) method.invoke(instance, arguments);
+        }
+        catch (InvocationTargetException ex) {
+            throw (Exception) ex.getCause();
+        }
+    }
+
+    /**
+     * Invokes a static private method for a class.
+     *
+     * @param clss the class whose static method to invoke
+     * @param methodToExecute the name of the method to invoke
+     * @param arguments the optional arguments
+     * @param <T> the type of the result
+     * @return the method's result
+     * @throws Exception if the method failed
+     * @noinspection ProhibitedExceptionThrown, ThrowInsideCatchBlockWhichIgnoresCaughtException,
+     *               unchecked
+     */
+    public static <T> T invokeStaticMethod(Class<?> clss,
+            String methodToExecute, Object... arguments) throws Exception {
+        final Method method = getClassDeclaredMethod(clss, methodToExecute, arguments.length);
+        try {
+            return (T) method.invoke(null, arguments);
+        }
+        catch (InvocationTargetException ex) {
+            throw (Exception) ex.getCause();
```

Comment:
Why not simply rethrow ?

Reply:
`ex.getCause()` returns `Throwable`, but none of invoked methods should throw `Trhowable` or `Error`.
Marking this method as `throws Throwable` requires many changes in the test code.

I added an assertion to ensure that no method throws `Throwable`.

Reply:
Can we wrap in illegal stat exception ?

Reply:
We can keep it as `InvocationTargetException`. This requires some changes in the test code. Done.